### PR TITLE
Fix: Add support for pending & maintenance in badges

### LINKED ISSUE #2456: Badge shows "Down" while maintenance mode is active
### ⚠️ Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### Description

When a Monitor is in Maintenance Mode, the status badge always displays that the monitor is down, even if it's currently up. 

### 👟 Reproduction steps

1. Create a maintenance for a monitor.
2. Activate the maintenance mode
3. The status badge of the selected monitor shows down or the set down-label

### 👀 Expected behavior

I thought it would display maintenance, n/a, or the correct status.

### 😓 Actual Behavior

while the monitor is in an active maintenance mode, its status badge displays down.

### 🐻 Uptime-Kuma Version

1.19.0

### 💻 Operating System and Arch

Ubuntu 18.04 64x

### 🌐 Browser

Google Chrome 108.0.5359.95 (Offizieller Build) (64-Bit)

### 🐋 Docker Version

20.10.21

### 🟩 NodeJS Version

_No response_

### 📝 Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by chakflying**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]: 
- [x] I have read and understand the pull request rules.

# Description

Fixes #2456

## Type of change

Please delete any options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [ ] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [ ] My changes generate no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)

image

### CODE REVIEW

**Review by mathiash98 — COMMENTED** [review]
Good addition, could perhaps merge the two switch statements and use ternary expression to set badgeLabel

File: server/routers/api-router.js [hunk 1]
```diff
@@ -139,11 +143,30 @@ router.get("/api/badge/:id/status", cache("5 minutes"), async (request, response
...
+                    break;
+                case 1:
+                    badgeValues.color = upColor;
+                    badgeValues.message = upLabel;
+                    break;
+                case 2:
+                    badgeValues.color = pendingColor;
+                    badgeValues.message = pendingLabel;
+                    break;
+                case 3:
```
Comment (Saibamen) [inline server/routers/api-router.js:162]: Can you use our key names? (PENDING, UP etc)

Comment (chakflying) [reply server/routers/api-router.js:162]: Good point, but this is merged already I'm not sure how should I handle this.

Comment (Saibamen) [reply server/routers/api-router.js:162]: New PR ;)

**Pull request merged**